Implemented edit distance algorithm. Case sensitive.

necrashter opened the pull request:
Used Wagner–Fischer algorithm, explained on wikipedia.

If you want comment lines tell me and I will add them tonight.
Also I can implement case insensitive version but it should be easy (just convert everything to upper or lower case)

necrashter committed:
Implemented edit distance algorithm. Case sensitive.

Used Wagner–Fischer algorithm, explained on wikipedia.

If you want comment lines tell me and I will add them tonight.
Also I can implement case insensitive version but it should be easy (just convert everything to upper or lower case)

manishrw commented on sorting and basics/edit_distance.py:
```diff
@@ -0,0 +1,20 @@
+def getEditDistance(str1, str2):
```
Also, add the unit test cases in the tests directory. Check the other test cases as a reference.

necrashter committed:
Added unit tests, doctest and comments

necrashter commented:
All the requested changes have been made. Please check again and provide feedback. Thanks.

manishrw commented:
Looks good. I've moved to pytest from unittest. This is to test and build after a feature addition / updation.
It would be great if you could also move the test case to pytest. You could use other test cases as reference. 
It's not a big change, please let me know if you will complete it else I could do it too.

necrashter committed:
Moved from unittest to pytest

necrashter committed:
Renamed and recategorized

necrashter committed:
Fixed encoding bug

necrashter commented:
Sorry for the delay. I replaced unittest with pytest. Also my code was broken because of an encoding issue; since i was using Python 3, while checks were using Python 2. I fixed it and passed the check.

By the way, I moved it to "misc" directory (I cannot import anything from "sorting and basics" directory because of the space in its name). Do you think it is appropriate? Or should I move it to somewhere else?

manishrw commented:
Looks good. I would suggest to keep it under dp dir. Also keep the test under "test" dir.

necrashter committed:
Moved to dp folder

necrashter commented:
I've just realized that you renamed the folder "tests" to "test". I have renamed too. Also I moved the module to "dp" directory.
